platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ImportMigrationDirectoryEntryImpl.java

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.configuration.migration;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.security.common.audit.SecurityLogger;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.filefilter.TrueFileFilter;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.migration.ImportMigrationEntry;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationReport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class provides an implementation of the {@link
+ * org.codice.ddf.migration.ImportMigrationEntry} representing a directory that was exported.
+ */
+@SuppressWarnings(
+    "squid:S2160" /* the base class equals() is sufficient for our needs. entries are unique based on their paths */)
```

Comment:
❗️ Please provide overridden `equals` and `hashcode` methods that invoke super instead of suppressing. It makes the intent clearer and will reduce the likelihood of a developer breaking the contract in the future.

Reply:
This was before your PR got merged. Not a fan but will do it